ui/src/shared/components/GaugeChart.js

```diff
@@ -0,0 +1,88 @@
+import React, {PropTypes, Component} from 'react'
+import shallowCompare from 'react-addons-shallow-compare'
+import lastValues from 'shared/parsing/lastValues'
+import Gauge from 'shared/components/Gauge'
+
+import {DEFAULT_COLORS} from 'src/dashboards/constants/gaugeColors'
+import {DASHBOARD_LAYOUT_ROW_HEIGHT} from 'shared/constants'
+
+class GaugeChart extends Component {
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState)
+  }
+
+  render() {
+    const {
+      data,
+      cellHeight,
+      isFetchingInitially,
+      colors,
+      resizeCoords,
+      resizerTopHeight,
+    } = this.props
+
+    // If data for this graph is being fetched for the first time, show a graph-wide spinner.
+    if (isFetchingInitially) {
+      return (
+        <div className="graph-empty">
+          <h3 className="graph-spinner" />
+        </div>
+      )
+    }
+
+    const lastValue = lastValues(data)[1]
+    const precision = 100.0
+    const roundedValue = Math.round(+lastValue * precision) / precision
+
+    // When a new height is passed the Gauge component resizes internally
+    // Passing in a new often ensures the gauge appears sharp
+
+    const initialCellHeight =
+      cellHeight && (cellHeight * DASHBOARD_LAYOUT_ROW_HEIGHT).toString()
+
+    const resizeCoordsHeight =
+      resizeCoords && (resizeCoords.h * DASHBOARD_LAYOUT_ROW_HEIGHT).toString()
+
+    const height = (resizeCoordsHeight ||
+      initialCellHeight ||
+      resizerTopHeight ||
+      300)
+      .toString()
+
+    return (
+      <div className="single-stat" ref={r => (this.gaugeContainer = r)}>
```

Comment:
Is this ref necessary? I think it can be removed.

Reply:
ah I think that's a vestige of a previous approach I was taking. Good catch!